Server/ConcordiaCurriculumManager/Services/CourseGroupingService.cs

```diff
@@ -222,4 +223,28 @@ public async Task DeleteCourseGroupingRequest(Guid dossierId, Guid requestId)
             throw new ServiceUnavailableException("The course grouping could not be deleted");
         }
     }
+
+    public async Task<CourseGrouping> PublishCourseGrouping(Guid commonIdentifier)
+    {
+        var newCourseGrouping = await _courseGroupingRepository.GetCourseGroupingByCommonIdentifier(commonIdentifier) ?? throw new NotFoundException($"The course grouping with common ID {commonIdentifier} was not found.");
```

Comment:
`GetCourseGroupingByCommonIdentifier` wouldn't work here because it doesn't query for deleted groupings. You'd need a new repository method that checks only for max non-null version without regards to state.

Reply:
You are right; I edited it